drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,330 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
 
 	return nil
 }
+
+// StartAndWaitForVMIMigration starts the VM migration and waits for the VM to be in running state in the new node
+func StartAndWaitForVMIMigration(virtualMachineCtx *scheduler.Context, ctx context1.Context, schedDriver scheduler.Driver) error {
+
+	vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{virtualMachineCtx})
+	if err != nil {
+		return err
+	}
+
+	// Get the namespace and name of the VM
+	if len(vms) == 0 {
+		return fmt.Errorf("no VMs found for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	}
+	vmiNamespace := vms[0].Namespace
+	vmiName := vms[0].Name
+	if len(vms) > 1 {
+		return fmt.Errorf("more than 1 VMs found for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	}
+
+	//Get the node where the vm is scheduled before the migration
+	nodeName, err := GetNodeOfVM(vms[0])
+	if err != nil {
+		return err
+	}
+	log.Infof("VM [%s] in namespace [%s] is scheduled on node [%s]", vmiName, vmiNamespace, nodeName)
+
+	// Start the VM migration
+	migration, err := kubevirtdy.Instance().CreateVirtualMachineInstanceMigration(ctx, vmiNamespace, vmiName)
+	if err != nil {
+		log.Errorf("Failed to create VM migration for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+		return err
+	}
+	log.Infof("VM migration created for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+
+	// wait for completion
+	var migr *kubevirtdy.VirtualMachineInstanceMigration
+
+	// get volumes from app context
+	volume, err := schedDriver.GetVolumes(virtualMachineCtx)
+	if err != nil {
+		return err
+	}
+
+	t := func() (interface{}, bool, error) {
+		migr, err = kubevirtdy.Instance().GetVirtualMachineInstanceMigration(ctx, vmiNamespace, migration.Name)
+		if err != nil {
+			return "", false, fmt.Errorf("failed to get migration for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+		}
+		if !migr.Completed {
+			return "", true, fmt.Errorf("waiting for migration to complete for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+		}
+
+		// wait until there is only one pod in the running state
+		testPod, err := getVMPod(virtualMachineCtx, volume[0])
+		if err != nil {
+			return "", true, err
+		}
+
+		//Get the node where the vm is scheduled after the migration
+		nodeNameAfterMigration := testPod.Spec.NodeName
```

Comment:
This could be a race condition in a heavily loaded setup. So Keep a timeout of 5 mins for the `nodeNameAfterMigration` has changed from previous name. If it has changed then we can assume safely that new target has been selected. If not, then 5 mins is the max time we will check for it.

Reply:
Done added 5 minute wait for node_name check

Reply:
@dbhatnagar-px : I don't understand your comment and why we have to wait some more after the migration object says that the migration is completed. At that point, we can simply return error if the node name has not changed. No retried are necessary. We should only keep retrying until the "migration" object says that the migration is still in progress (we are already retrying this). That should take care of longer wait time in a loaded setup.

Reply:
You are right Neelesh. We have removed this check from here. Thanks.